src/main/java/net/hypixel/nerdbot/curator/Curator.java

```diff
@@ -1,419 +1,46 @@
-
-                if (!dev) {
-                    realPositive = discountBotAndUserReactions(group, message, positive);
-                    realNegative = discountBotAndUserReactions(group, message, negative);
-                }
-
-                if (realPositive == 0 && realNegative == 0) {
-                    log("[" + group.getName() + "] [" + message.getId() + "] Skipping because it has no positive and negative reactions!");
-                    continue;
-                }
-
-                if (realPositive < NerdBotApp.getBot().getConfig().getMinimumThreshold()) {
-                    log("[" + group.getName() + "] [" + message.getId() + "] Skipping because it has less than the minimum threshold of " + NerdBotApp.getBot().getConfig().getMinimumThreshold() + " positive reactions!");
-                    continue;
-                }
-
-                getAndUpdateUserReactions(group, message, positive, negative);
-
-                if (message.getReaction(greenlit) != null) {
-                    checkGreenlit(group, message, realPositive, realNegative);
-                    continue;
-                }
-
-                double ratio = getRatio(realPositive, realNegative);
-                double requiredRatio = NerdBotApp.getBot().getConfig().getPercentage();
-                String ratioMessage = "[" + group.getName() + "] [" + message.getId() + "] Reaction ratio is " + Util.DECIMAL_FORMAT.format(ratio) + "% (" + realPositive + " positive / " + realNegative + " negative)";
-                if (ratio < requiredRatio) {
-                    ratioMessage += ". This is below the minimum threshold of " + Util.DECIMAL_FORMAT.format(requiredRatio) + "%!";
-                    log(ratioMessage);
-                    continue;
-                }
-
-                if (readOnly) {
-                    log("[" + group.getName() + "] [" + message.getId() + "] Skipping logging the message because this is a read-only run!");
-                    continue;
-                }
-
-                ratioMessage += ". Greenlighting this message!";
-                log(ratioMessage);
-
-                message.addReaction(greenlit).queue();
-                GreenlitMessage msg = createGreenlitMessage(group, message, realPositive, realNegative);
-                sendGreenlitToChannel(submissionChannel, group, msg);
-            }
-        }
-
-        if (!users.isEmpty()) Database.getInstance().updateUsers(users);
-
-        long end = System.currentTimeMillis();
-        elapsed = end - start;
-
-        log(Util.DASHED_LINE);
-        log("Curating process finished at " + new Date(end) + ". Elapsed time: " + elapsed + "ms (" + Time.formatMs(elapsed) + ")");
-        log(Util.DASHED_LINE);
-
-        logChannel.sendMessage("Curating process finished at " + new Date(end) + ". Elapsed time: " + elapsed + "ms (" + Time.formatMs(elapsed) + ")").queue();
+        return (double) positiveReactions / (positiveReactions + negativeReactions) * 100;
```

Comment:
this will cast the result of integer division to a double- you should cast positiveReactions and negativeReactions to double before you perform the return statement logic